lib/spack/llnl/util/cpu/microarchitectures.json

```diff
@@ -1161,6 +1161,97 @@
         }
       }
     },
+    "thunderx2": {
+      "from": "aarch64",
+      "vendor": "0x43",
+      "features": [
+        "fp",
+        "asimd",
+        "evtstrm",
+        "aes",
+        "pmull",
+        "sha1",
+        "sha2",
+        "crc32",
+        "atomics",
+        "cpuid",
+        "asimdrdm"
+      ],
+      "compilers": {
+        "gcc": [
+          {
+            "versions": "4.8:4.8.9",
+            "flags": "-march=armv8-a"
+          },
+          {
+            "versions": "4.9:5.9",
+            "flags": "-march=armv8-a+crc+crypto"
+          },
+          {
+            "versions": "6:6.9",
+            "flags": "-march=armv8.1-a+crc+crypto"
+          },
+          {
+            "versions": "7:",
+            "flags": "-mcpu=thunderx2t99"
+          }
+        ],
+        "clang": {
+          "versions": ":",
+          "flags": "-march=armv8-a -mcpu=generic"
+        }
+      }
+    },
+    "a64fx": {
+      "from": "aarch64",
+      "vendor": "0x46",
+      "features": [
+        "fp",
+        "asimd",
+        "evtstrm",
+        "aes",
+        "pmull",
+        "sha1",
+        "sha2",
+        "crc32",
+        "atomics",
+        "cpuid",
+        "asimdrdm",
+        "fphp",
+        "asimdhp",
+        "fcma",
+        "dcpop",
+        "sve"
+      ],
+      "compilers": {
+        "gcc": [
+          {
+            "versions": "4.8:4.8.9",
+            "flags": "-march=armv8-a"
+          },
+          {
+            "versions": "4.9:5.9",
+            "flags": "-march=armv8-a+crc+crypto"
+          },
+          {
+            "versions": "6:6.9",
+            "flags": "-march=armv8.1-a+crc+crypto"
+          },
+          {
+            "versions": "7:7.9",
+            "flags": "-arch=armv8.2a+crc+crypt+fp16"
+          },
+          {
+            "versions": "8:",
+            "flags": "-arch=armv8.2a+crc+aes+sh2+fp16+sve -msve-vector-bits=512"
```

Comment:
Curious why `crypto` was omitted here but not above.

Reply:
In gcc 8 changes (https://gcc.gnu.org/gcc-8/changes.html):
* The Armv8-A +crypto extension has now been split into two extensions for finer grained control: 
    * +aes which contains the Armv8-A AES crytographic instructions.
    * +sha2 which contains the Armv8-A SHA2 and SHA1 cryptographic instructions.

Reply:
From what I understand reading the man pages of GCC 8 and what you linked `crypto` is still available, and it is equivalent to setting `+aes+sha2+simd`. 

Pasting the snippet I read below for ease of reference:
```
       crypto
           Enable Crypto extension.  This also enables Advanced SIMD and floating-point instructions.

       fp  Enable floating-point instructions.  This is on by default for all possible values for options -march and -mcpu.

       simd
           Enable Advanced SIMD instructions.  This also enables floating-point instructions.  This is on by default for all possible values
           for options -march and -mcpu.

       sve Enable Scalable Vector Extension instructions.  This also enables Advanced SIMD and floating-point instructions.

       lse Enable Large System Extension instructions.  This is on by default for -march=armv8.1-a.

       rdma
           Enable Round Double Multiply Accumulate instructions.  This is on by default for -march=armv8.1-a.

       fp16
           Enable FP16 extension.  This also enables floating-point instructions.

       fp16fml
           Enable FP16 fmla extension.  This also enables FP16 extensions and floating-point instructions. This option is enabled by default
           for -march=armv8.4-a. Use of this option with architectures prior to Armv8.2-A is not supported.

       rcpc
           Enable the RcPc extension.  This does not change code generation from GCC, but is passed on to the assembler, enabling inline asm
           statements to use instructions from the RcPc extension.

       dotprod
           Enable the Dot Product extension.  This also enables Advanced SIMD instructions.

       aes Enable the Armv8-a aes and pmull crypto extension.  This also enables Advanced SIMD instructions.

       sha2
           Enable the Armv8-a sha2 crypto extension.  This also enables Advanced SIMD instructions.

       sha3
           Enable the sha512 and sha3 crypto extension.  This also enables Advanced SIMD instructions. Use of this option with architectures
           prior to Armv8.2-A is not supported.

sm4 Enable the sm3 and sm4 crypto extension.  This also enables Advanced SIMD instructions.  Use of this option with architectures
           prior to Armv8.2-A is not supported.

       Feature crypto implies aes, sha2, and simd, which implies fp.  Conversely, nofp implies nosimd, which implies nocrypto, noaes and
       nosha2.
```

Reply:
```suggestion
            "flags": "-arch=armv8.2a+crc+aes+sha2+fp16+sve -msve-vector-bits=512"
```